components/Layout.js

```diff
@@ -1,15 +1,25 @@
-import Nav from './Nav';
+import { useRouter } from 'next/router';
+import { heroOptions } from '../utils/hero-options';
+import Hero from './Hero';
 import Meta from './Meta';
 import Footer from './Footer';
 import styles from '../styles/Layout.module.scss';
 
 export default function Layout({ children }) {
+  const router = useRouter();
+  const heroPathKeys = Object.keys(heroOptions)
```

Comment:
is this block of code (sort and filter) for something else? 
`<Hero {...heroOptions[router.pathname]} />` works 

```suggestion
  const router = useRouter();
  return (
    <>
      <Meta />
      <main className={styles.main}>
        <Hero {...heroOptions[router.pathname]} />
        {children}
        <Footer />
      </main>
    </>
  );
```

Reply:
Yes, that part is needed there, @cherylli .
It is a rule created to make sure all the future blog pages are initially directed to the primary blog hero we see on line 26 of `hero-options.js`
On line 10, I'm grabbing the object keys so that the object elements are always sorted out and filtered by their initial part ( I recommend consoling log the "k" on line 12). 
Then, on line 14, you see the heroKey part to be considered will always be **the last** part of that object key in case I want to overwrite the condition by adding a new "blog" property with a more specific path, such as `blog/my-post-title`.